fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -5240,6 +5231,25 @@ public ChangedTransactionDetail updateDisbursementDateAndAmountForTranche(final
         return changedTransactionDetail;
     }
 
+    public BigDecimal getPrincipalAmountForRepaymentSchedule() {
+        BigDecimal principalAmount = BigDecimal.ZERO;
+
+        if (isMultiDisburmentLoan() && isDisbursed()) {
```

Comment:
@josehernandezfintecheandomx I think this logic is faulty if the loan is not a multi disbursement loan... 

It should rather use the totalDisbursedAmount and no check whether it is multi disbursement or not... 

What do you think?

Reply:
Also, i am a little bit hesitant here...if expected disbursement details were provided, we should rather show repayment schedule based on those total principal, no? @bharathcgowda

Reply:
Done for the Integration Test